internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -580,14 +580,15 @@ func ComposedFieldOwnerName(xr *composite.Unstructured) string {
 // any existing composed resources from the API server. It also loads their
 // connection details.
 type ExistingComposedResourceObserver struct {
-	resource client.Reader
-	details  managed.ConnectionDetailsFetcher
+	resource   client.Reader
+	ncresource client.Reader
```

Comment:
I'd suggest cached and uncached here also now that there's two.

Reply:
Done